src/sdk/analytics/hooks/useIntersectionObserver.ts

```diff
@@ -0,0 +1,34 @@
+import type { RefObject } from 'react'
+import { useEffect, useState } from 'react'
+
+interface Args extends IntersectionObserverInit {
+  freezeOnceVisible?: boolean
+}
+
+function useIntersectionObserver(
```

Comment:
Why implementing this when we already use a lib for that?
https://github.com/vtex-sites/base.store/blob/6245be5b3eb7c63e3345e29960d9d0711d165ab1/src/sdk/search/PageBreak.tsx#L2

Reply:
I didn't know this. I will use this lib.